rcl/include/rcl/context.h

```diff
@@ -240,6 +240,32 @@ RCL_WARN_UNUSED
 rcl_context_instance_id_t
 rcl_context_get_instance_id(rcl_context_t * context);
 
+/// Returns the context domain id.
+/**
+ * \pre If context is uninitialized, then it is undefined behavior.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | Yes [1]
+ * Uses Atomics       | No
+ * Lock-Free          | No
+ *
+ * <i>[1] Calling the function asynchronously with `rcl_init` or `rcl_shutdown` can result
+ *  in the function sometimes succeeding and sometimes returning `RCL_RET_INVALID_ARGUMENT`.
```

Comment:
@ivanpauno Hmm, not sure if went through this before, but this doesn't seem to imply it's safe.

Reply:
See discussion in https://github.com/ros2/rcl/pull/718#discussion_r456521461.

Reply:
Alright, yeah, it's unusual but not unsafe.